libraries/chain/webassembly/runtimes/eos-vm-oc/code_cache.cpp

```diff
@@ -136,11 +137,11 @@ const code_descriptor* const code_cache_async::get_descriptor_for_code(const dig
    //check for entry in cache
    code_cache_index::index<by_hash>::type::iterator it = _cache_index.get<by_hash>().find(boost::make_tuple(code_id, vm_version));
    if(it != _cache_index.get<by_hash>().end()) {
-      if (is_main_thread())
+      if (is_write_window)
          _cache_index.relocate(_cache_index.begin(), _cache_index.project<0>(it));
       return &*it;
    }
-   if(!is_main_thread()) // on read-only thread
+   if(!is_write_window)
       return nullptr;
```

Comment:
one initial concern I had was what would happen for a contract _only_ used for ROtrxs -- would it never get tiered up? @heifner pointed out to me that we do run ROtrxs during the write window as a low priority task, so it seems like a contract being used only for ROtrxs would indeed have an opportunity to be tiered up during that period

Reply:
Thanks for digging this out. I had another concern last night when re-reading the code and was going to ask today. After a ROtrx's contract is evicted, how is it got back into the code cache again if it happens to be scheduled to run in the read window? Where is the code to make it wait until write window? Thanks again.

Reply:
I merge the PR now. Will do further work if necessary.